GVRf/Framework/framework/src/main/java/org/gearvrf/GVRSpotLight.java

```diff
@@ -115,4 +140,66 @@ public float getOuterConeAngle() {
     public void setOuterConeAngle(float angle) {
         setFloat("outer_cone_angle", (float) Math.cos(Math.toRadians(angle)));
     }
+    
+    /**
+     * Updates the position, direction and shadow matrix
+     * of this light from the transform of scene object that owns it.
+     * The shadow matrix is the model/view/projection matrix
+     * from the point of view of the light.
+     */
+    public void onDrawFrame(float frameTime)
+    {
+        GVRSceneObject parent = owner;
+        float[] odir = getVec3("world_direction");
+        float[] opos = getVec3("world_position");
+        Matrix4f worldmtx = parent.getTransform().getModelMatrix4f();
+        Matrix4f lightrot = new Matrix4f();
+        Vector3f olddir = new Vector3f(odir[0], odir[1], odir[2]);
```

Comment:
You may want to switch to using member variables for these vectors and matrices and updating them instead of instantiating new ones. This is applicable to other methods too. It can be done later in an optimization round.

The new instances on every frame would create a bit more work for the garbage collector. Nothing noticeable but seems easy enough to be addressed.